app/background_tasks/refresh_course_task.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'rust_langs_cli_executor'
+
+class RefreshCourseTask
+  def initialize
+  end
+
+  def run
+    CourseRefresh.all.each do |task|
+      finished_successfully = false
+      begin
+        courses = Course.where(course_template_id: task.course_template_id)
+        Rails.logger.info("Refreshing courses created from template #{task.course_template_id}")
+        courses.each do |course|
+          Rails.logger.info("Refreshing course #{course.name}")
+          RustLangsCliExecutor.refresh(course, task.id, task.no_background_operations, task.no_directory_changes)
+          finished_successfully = true
+        end
+      rescue => e
+        Rails.logger.error("Task failed: #{e}")
+        Rails.logger.error(e.backtrace.join("\n"))
+      end
+      task.destroy! if finished_successfully
```

Comment:
Destroys entry created in database and all the phases.

Reply:
Yeah maybe someting like `.update(completed: true)` would be better here.

Reply:
Yea, I will use the status from the CourseRefresh, i.e. Course.where(status: :not_started).each do |t|...